API.md

```diff
@@ -1,4 +1,111 @@
-# Authmosphere API documentation
+# Authmosphere API Documentation
+
+## OAuth Tooling
+
+This tooling provides helper functions to request and validate tokens based on the OAuth 2.0 [RFC 6749](https://tools.ietf.org/html/rfc6749) specification.
+
+### getAccessToken
+
+Requests a token based on the given configuration (which specifies the grant type and corresponding parameters). See the [`OAuthConfig` documentation](#tbd) for details.
+
+#### Usage
+
+```ts
+import {
+  OAuthGrantType,
+  ClientCredentialsGrantConfig,
+  Token,
+  getAccessToken
+} from 'authmosphere';
+
+const config: ClientCredentialsGrantConfig = {
+  grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
+  credentialsDir: './crendentials',
+  accessTokenEndpoint: 'example.com/token',
+  scopes: ['my-app.read', 'my-app.write'];
+};
+
+getAccessToken(config)
+  .then((token: Token) => {
+    // ...use the token...
+  })
+  .catch((err) => {
+    // ...handle the error...
+  });
+```
+
+#### Signature
+
+`getAccessToken(config[, logger]): Promise<Token>`
+
+#### Arguments
+
+* [`config: OAuthConfig`](#tbd) - OAuth configuration for the request (specify grant type and corresponding parameters)
+* [`logger?: Logger`](#tbd) - Optional logger
+
+#### Returns
+
+`Promise<Token>`
+
+### getTokenInfo
+
+Requests validation information from the specified `tokenInfoUrl` and returns a `Promise` which resolves with these information, if the token is valid. Otherwise, it rejects with an error.
+
+#### Usage
+
+```ts
+import {
+  Token,
+  getTokenInfo
+} from 'authmosphere';
+
+getTokenInfo('example.com/tokeninfo', '1234-5678-9000')
+  .then((token: Token) => {
+    // ...token is valid...
+  })
+  .catch((err) => {
+    // ...token is invalid...
+  })
+```
+
+#### Signature
+
+`getTokenInfo<T>(tokenInfoUrl, accessToken[, logger]): Promise<Token<T>>`
+
+#### Arguments
+
+* `tokenInfoUrl: string` - OAuth endpoint for validating tokens
+* `accessToken: string` - Token to be validated
+* [`logger?: Logger`](#tbd) - Optional logger
+
+#### Returns
+
+`Promise<Token>`
+
+### createAuthCodeRequestUri
+
+Helper function to request an authorization code when using the [Authorization Code Grant](https://tools.ietf.org/html/rfc6749#page-24).
+
+#### Usage
+
+```ts
+const uri = createAuthCodeRequestUri('example.com/authorize', 'http://your-app.com/handle-auth-code', '1234-client-id');
+```
+
+#### Signature
+
+`createAuthCodeRequestUri(authorizationEndpoint, redirectUri, clientId[, queryParams]): string`
+
+#### Arguments
+
+* `authorizationEndpoint: string` - [OAuth authorization endpoint](https://tools.ietf.org/html/rfc6749#page-18)
+* `redirectUri: string` - URI specifying the endpoint the authorization code is responded to (see [OAuth 2.0 specification](https://tools.ietf.org/html/rfc6749#section-3.1.2) for details)
```

Comment:
Would add here `Absolute URL`?

Reply:
Instead of URI?

Reply:
Sorry meant `Absolute URI` at least that is what the link says

Reply:
👍